napalm/eos/eos.py

```diff
@@ -156,11 +177,15 @@ def open(self):
             )
 
             if self.device is None:
-                self.device = pyeapi.client.Node(connection, enablepwd=self.enablepwd)
+                self.device = Node(connection, enablepwd=self.enablepwd)
             # does not raise an Exception if unusable
 
-            # let's try to run a very simple command
-            self.device.run_commands(["show clock"], encoding="text")
+            # let's try to determine if we need to use new EOS cli syntax
+            cli_version = self._determine_syntax_version(
+                self.device.run_commands(["show version"], encoding="text")[0]["output"]
```

Comment:
I think the reasoning why we had `self.device.run_commands(["show clock"], encoding="text")` as text, was because `show clock` is only available in that format. That said, I _think_ doing `self.device.run_commands(["show version"])` could simplify the code by getting rid of the regular expression complication and do a simple lookup under the `version` key in the return, i.e.,

```python
sh_ver = self.device.run_commands(["show version"])
cli_version = 2 if EOSVersion(sh_ver[0]['version']) >= EOSVersion("4.23.0") else 1
```